lib/logstash/filters/mutate.rb

```diff
@@ -266,9 +266,13 @@ def convert(event)
       if original.nil?
         next
       elsif original.is_a?(Hash)
-        @logger.debug("I don't know how to type convert a hash, skipping",
-                      :field => field, :value => original)
-        next
+        #@logger.debug("I don't know how to type convert a hash, skipping",
+        #              :field => field, :value => original)
+        #next
+        value = original
+        original.each do |hfield, hvalue|
```

Comment:
kv filter can preobrazovіvat all options hash and I think it will be
convenient to use this design as a hash so that in the resulting hash
values are all given to the type of data I need.

That's part of my config
...
filter {
  grok {
    match => [ "message", "%{IP:client} \[%{HTTPDATE:timestamp}\]
\"%{WORD:verb} %{URIPATHPARAM:request} HTTP/%{NUMBER:httpversion}\"
ui\:%{BASE16NUM:ui}" ]
  }
  urldecode {
      field => "request"
  }
  kv {
      field_split => "&?"
      source => "request"
      target => "args"
  }
  mutate {
    remove_field => [ "httpversion", "message", "verb", "path", "host",
"@version", "request" ]
    convert => {"args" => "integer"}
  }
}
...

I thought that this design may be necessary not only to me. I do not know
how to program in ruby but I have everything working.


2015-09-09 12:49 GMT+03:00 Pere Urbón <notifications@github.com>:

> In lib/logstash/filters/mutate.rb
> <https://github.com/logstash-plugins/logstash-filter-mutate/pull/51#discussion_r39024738>
> :
>
> > @@ -266,9 +266,13 @@ def convert(event)
> >        if original.nil?
> >          next
> >        elsif original.is_a?(Hash)
> > -        @logger.debug("I don't know how to type convert a hash, skipping",
> > -                      :field => field, :value => original)
> > -        next
> > +        #@logger.debug("I don't know how to type convert a hash, skipping",
> > +        #              :field => field, :value => original)
> > +        #next
> > +        value = original
> > +        original.each do |hfield, hvalue|
>
> you can he each_pair here, check
> http://ruby-doc.org/core-2.2.0/Hash.html#method-i-each_pair
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/logstash-plugins/logstash-filter-mutate/pull/51/files#r39024738>
> .
>



-- 
---
С уважением, Олег Черний,
руководитель отдела разработки  AUTO.RIA.com

RIA.com
тел./факс.: 0 432 555-200 (многоканальний)
моб: 0 (67) 295-27-52
E-mail: *oleg.cherniy@ria.ua <oleg.cherniy@ria.ua>*